core/engine.go

```diff
@@ -157,8 +141,7 @@ func (e *Engine) Init(globalCtx, runCtx context.Context) (run func() error, wait
 // and that the remaining metrics samples in the pipeline should be processed as the background
 // process is about to exit.
 func (e *Engine) startBackgroundProcesses(
-	globalCtx, runCtx context.Context, execRunResult, engineRunResult chan error,
-	runSubCancel func(), processMetricsAfterRun chan struct{},
+	globalCtx context.Context, processMetricsAfterRun chan struct{},
```

Comment:
This is a very nice simplification :+1:

Reply:
Just wait for https://github.com/grafana/k6/pull/2815 :sweat_smile: